datafusion/core/src/datasource/file_format/mod.rs

```diff
@@ -149,6 +154,11 @@ impl DefaultFileType {
             file_format_factory,
         }
     }
+
+    /// get a [FileFormatFactory] struct
+    pub fn get_format_factory(&self) -> Arc<dyn FileFormatFactory> {
```

Comment:
Another more 'rust idomatic' way to do this might be a method like this (which would allow the user to choose to copy if they wanted or use the reference)

```rust
    /// get a reference to the inner [FileFormatFactory] struct
    pub fn as_format_factory(&self) -> &Arc<dyn FileFormatFactory> {
        &self.file_format_factory
    }
```

Reply:
got it